pkg/syncer/mutators/interface_test.go

```diff
@@ -0,0 +1,96 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mutators
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+func TestMutators(t *testing.T) {
+	// Register the mutator.
+	tdm := &TestDeploymentMutator{}
+
+	deployment := appsv1.Deployment{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Deployment",
+			APIVersion: "apps/v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Name: "test-deployment",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: new(int32),
+		},
+		Status: appsv1.DeploymentStatus{
+			ObservedGeneration: int64(0),
+		},
+	}
+
+	// Convert to unstructured.
+	syncedObject := &unstructured.Unstructured{}
+	err := convert(&deployment, syncedObject)
+	if err != nil {
+		t.Errorf("convert failed: %v", err)
+	}
+
+	// Apply mutators to the object.
+	err = tdm.Mutate(syncedObject)
+	if err != nil {
+		t.Errorf("Mutate() = %v", err)
+	}
+
+	// Convert the unstructured object back to a deployment.
+	err = runtime.DefaultUnstructuredConverter.FromUnstructured(syncedObject.UnstructuredContent(), &deployment)
```

Comment:
(No action required) Maybe separate out the conversion to/from unstructured from the mutation function to simplify testing? 

```golang
def mutate(obj *unstructured.Unstructured) error {
  deployment := unmarshal(...)
  err := updateDeployment(deployment)
  if err != nil {
   return err
  } 
  runtime.DefaultUnstructuredConverter.FromUnstructured(obj.UnstructuredContent(), deployment)
}
```

And then target `updateDeployment` for testing of mutation changes.